sycl/source/detail/graph_impl.cpp

```diff
@@ -1775,6 +1844,208 @@ void dynamic_parameter_base::updateAccessor(
+    auto DynCG = DynCGInfo.DynCG.lock();
+    if (DynCG) {
+      auto &CG = DynCG->MKernels[DynCGInfo.CGIndex];
+      dynamic_parameter_impl::updateCGAccessor(CG, DynCGInfo.ArgIndex, Acc);
+    }
+  }
+
+  std::memcpy(MValueStorage.data(), Acc,
+              sizeof(sycl::detail::AccessorBaseHost));
+}
+
+void dynamic_parameter_impl::updateCGArgValue(
+    std::shared_ptr<sycl::detail::CG> CG, int ArgIndex, const void *NewValue,
+    size_t Size) {
+  auto &Args = static_cast<sycl::detail::CGExecKernel *>(CG.get())->MArgs;
+  for (auto &Arg : Args) {
+    if (Arg.MIndex != ArgIndex) {
+      continue;
+    }
+    assert(Arg.MSize == static_cast<int>(Size));
+    // MPtr may be a pointer into arg storage so we memcpy the contents of
+    // NewValue rather than assign it directly
+    std::memcpy(Arg.MPtr, NewValue, Size);
+    break;
+  }
+}
+
+void dynamic_parameter_impl::updateCGAccessor(
+    std::shared_ptr<sycl::detail::CG> CG, int ArgIndex,
+    const sycl::detail::AccessorBaseHost *Acc) {
+  auto &Args = static_cast<sycl::detail::CGExecKernel *>(CG.get())->MArgs;
+
+  auto NewAccImpl = sycl::detail::getSyclObjImpl(*Acc);
+  for (auto &Arg : Args) {
+    if (Arg.MIndex != ArgIndex) {
+      continue;
+    }
+    assert(Arg.MType == sycl::detail::kernel_param_kind_t::kind_accessor);
+
+    // Find old accessor in accessor storage and replace with new one
+    if (static_cast<sycl::detail::SYCLMemObjT *>(NewAccImpl->MSYCLMemObj)
+            ->needsWriteBack()) {
+      throw sycl::exception(
+          make_error_code(errc::invalid),
+          "Accessors to buffers which have write_back enabled "
+          "are not allowed to be used in command graphs.");
+    }
+
+    // All accessors passed to this function will be placeholders, so we must
+    // perform steps similar to what happens when handler::require() is
+    // called here.
+    sycl::detail::Requirement *NewReq = NewAccImpl.get();
+    if (NewReq->MAccessMode != sycl::access_mode::read) {
+      auto SYCLMemObj =
+          static_cast<sycl::detail::SYCLMemObjT *>(NewReq->MSYCLMemObj);
+      SYCLMemObj->handleWriteAccessorCreation();
+    }
+
+    for (auto &Acc : CG->getAccStorage()) {
+      if (auto OldAcc = static_cast<sycl::detail::AccessorImplHost *>(Arg.MPtr);
+          Acc.get() == OldAcc) {
+        Acc = NewAccImpl;
+      }
+    }
+
+    for (auto &Req : CG->getRequirements()) {
+      if (auto OldReq = static_cast<sycl::detail::AccessorImplHost *>(Arg.MPtr);
+          Req == OldReq) {
+        Req = NewReq;
+      }
+    }
+    Arg.MPtr = NewAccImpl.get();
+    break;
+  }
+}
+
+dynamic_command_group_impl::dynamic_command_group_impl(
+    const command_graph<graph_state::modifiable> &Graph)
+    : MGraph{sycl::detail::getSyclObjImpl(Graph)}, MActiveCGF(0) {}
+
+void dynamic_command_group_impl::finalizeCGFList(
+    const std::vector<std::function<void(handler &)>> &CGFList) {
+  // True if kernels use sycl::nd_range, and false if using sycl::range
```

Comment:
Comment doesn't seem to apply to this code, should it be somewhere else or removed?